pkg/wasm/cache.go

```diff
@@ -137,6 +139,12 @@ func NewLocalFileCache(dir string, purgeInterval, moduleExpiry time.Duration, in
 		stopChan:           make(chan struct{}),
 		insecureRegistries: sets.New(insecureRegistries...),
 	}
+
+	// If the list of the given insecure registries is "*", then allow all the insecure registries.
+	if len(insecureRegistries) == 1 && insecureRegistries[0] == "*" {
```

Comment:
this worth a release notes, maybe a separate PR.

Reply:
Included in this PR. I have changed slightly to check if `insecureRegistries` contains `*`.

Reply:
it would be better to separate feature and tests. this feature can be landing in v1.14, WDYT?

Reply:
cherrypick for v1.14 sounds good to me.

https://github.com/istio/istio/pull/39024 is the PR for this feature.
Will rebase this PR after https://github.com/istio/istio/pull/39024 is merged.